packages/@aws-cdk/aws-s3/test/test.bucket.ts

```diff
@@ -72,6 +72,120 @@ export = {
     test.done();
   },
 
+  'valid bucket names'(test: Test) {
+    const stack = new cdk.Stack();
+
+    test.doesNotThrow(() => new s3.Bucket(stack, 'MyBucket1', {
+      bucketName: 'abc.xyz-34ab'
+    }));
+
+    test.doesNotThrow(() => new s3.Bucket(stack, 'MyBucket2', {
+      bucketName: '124.pp--33'
+    }));
+
+    test.done();
+  },
+
+  'bucket validation skips tokenized values'(test: Test) {
+    const stack = new cdk.Stack();
+
+    test.doesNotThrow(() => new s3.Bucket(stack, 'MyBucket', {
+      bucketName: new cdk.Token(() => '_BUCKET').toString()
+    }));
+
+    test.done();
+  },
+
+  'fails if bucket name has less than 3 or more than 63 characters'(test: Test) {
```

Comment:
Can you please add another failure test that asserts against the entire message? (especially good for CRs).

Reply:
Added and updated PR.